core/src/main/scala/cats/collections/Predicate.scala

```diff
@@ -2,16 +2,35 @@ package cats.collections
 
 import cats._
 
+private object PredicateHelpers {
+  // use the static Eval instances to reduce memory footprint
+  @inline def boolToNow(b: Boolean) = if (b) Eval.True else Eval.False
+
+  trait FromApplyF[A] extends Predicate[A] {
+    override def applyF(a: A): Eval[Boolean] =
+      throw new NotImplementedError("applyF implementation missing")
```

Comment:
is this needed? Why not just leave it abstract? I guess because otherwise you have an infinite loop and you want a cleaner error? If so, can you add a comment to that effect?

Reply:
It seems it cant be made to be abstract again because it is concrete in Predicate. I tried `abstract override` too but to no avail.